### DISCUSSION TITLE
DynamicLayout add column not working

### DISCUSSION THREAD

**Discussion by visualizor** [picoe/Eto#1912] [Q&A] [answered]
add an `ImageView` and a `StackLayout` onto a `DynamicLayout` using the following (before eventually adding this `DynamicLayout` to a `TabPage`) 
```
foreach (Control c in added)
{
    dyna.BeginVertical();
    dyna.AddAutoSized(c);
    dyna.EndVertical();
}
```
But I get this
![image](https://user-images.githubusercontent.com/25107288/111929596-363e3380-8a8d-11eb-9cf3-432122f4d0de.png)

Still seems to be in a single column rather than two separate ones.

**Comment by cwensley** [comment]
I'm not quite sure I understand what you are trying to accomplish.. which controls are in the "added" collection?  what do you want it to appear like?

To create columns, you need to use `BeginHorizontal()`.

**Reply by visualizor** [reply]
![image](https://user-images.githubusercontent.com/25107288/115962810-0dc1b300-a4eb-11eb-93a3-4c79904ea2d5.png)![image](https://user-images.githubusercontent.com/25107288/115962817-174b1b00-a4eb-11eb-8752-131a1082780f.png)

I want to do above. so the first control is the graph (`ImageView`) and the second control is the legend (`StackLayout` containing `Labels`)
I've tried `BeginHorizontal()` but got same result.

**Reply by cwensley** [reply]
Hm, the code you posted didn't show any of those controls.. however, maybe this might help point you in the right direction:

```
dyna.BeginHorizontal();

dyna.Add(myGraph);

// create vertical section for legend, which comes after the graph horizontally since we are in BeginHorizontal()
dyna.BeginVertical();
foreach (var legendControl in legendControls)
{
  dyna.Add(legendControl);
}
dyna.EndVertical();

dyna.EndHorizontal();
```

Alternatively, use `StackLayout`, which is sometimes easier:

```

// create legend
var myLegend = new StackLayout();
foreach (var legendControl in legendControls)
{
  myLegend.Items.Add(legendControl);
}

// create graph + legend beside it
var horizontalStack = new StackLayout { 
  Orientation = Orientation.Horizontal,
  VerticalContentAlignment = VerticalAlignment.Stretch,
  Items = { myGraph, new StackLayoutItem(myLegend, VerticalAlignment.Bottom) }
};

// place horizontalStack somewhere.

**Comment by visualizor** [comment, marked as answer]
Thanks @cwensley ! What worked was bracketing all the vertical additions with a top level `BeginiHorizontal()` and end, just like in your first example code. I wish this was in the doc.
